WordPress/Classes/Services/PostService.swift

```diff
@@ -0,0 +1,599 @@
+import Foundation
+
+class PostService : LocalCoreDataService {
+    static let ErrorDomain = "PostServiceErrorDomain"
+    static let DefaultNumberToSync: UInt = 40
+
+    enum PostType: String {
+        case any = "any"
+        case post = "post"
+        case page = "page"
+
+        func type() -> AbstractPost.Type {
+            switch self {
+            case .post: return Post.self
+            case .page: return Page.self
+            case .any: return AbstractPost.self
+            }
+        }
+    }
+
+    typealias SuccessPosts = ([AbstractPost]) -> Void
+    typealias FailureBasic = (NSError?) -> Void
```

Comment:
Probably drop the word "Basic" here as it doesn't really contribute to the reusability. If we end up changing this block we would then have to rename it from "Basic".